controllers/index.js

```diff
@@ -10,10 +10,47 @@ router.get('/', async (req, res, next) => {
     // Fetch templates
     const {templates} = await model.list(0, null);
     const filterOptions = {
-      sort: req.query.sort || 'views',
-      templateTypes: (req.query.templateTypes || 'all').split(','),
-      categories: (req.query.categories || 'all').split(',')
+      sort: (req.query.sort) ? req.query.sort.split(',') : ['views'],
+      templateTypes: (req.query.templateTypes) ? req.query.templateTypes.split(',') : ['all'],
+      categories: (req.query.categories) ? req.query.categories.split(',') : ['all']
     };
+    
+    // Sanitizing refining values
+    // Template Types
+    filterOptions.templateTypes.forEach(function(e,i){
+        if(Object.keys(enums.allowedFilterValues.templateTypes).indexOf(e)===-1) filterOptions.templateTypes.splice(i,1);
+    });
+    // User may remove the filter value manually and key will still exist, set default value
+    if(filterOptions.templateTypes.length===0) filterOptions.templateTypes = ['all'];
+
```

Comment:
Can this ever happen as you have

```
templateTypes: (req.query.templateTypes) ? req.query.templateTypes.split(',') : ['all'],
```

earlier? If templateTypes has length 0 (i.e. empty string), it would set it to ['all'], which means this check should be redundant.

Reply:
if we have  /?type=& , req.query.templateTypes will be true, then the split will thrown an error

Reply:
No it won't. Did you try? :)

An empty string is falsy, so ?type= without a proper value will not pass the ternary check and will thus fall back to ['all'].

Reply:
An empty string is falsy in JS, so ?type= would set req.query.templateTypes to "", which will not pass the ternary, and templateTypes will fall back to ['all'].

Reply:
![image](https://user-images.githubusercontent.com/1494564/60117386-44590600-977a-11e9-9d46-6db0305d6195.png)

Reply:
meh problems relies somewhere else ( i switched all values to be an array for consistence ), i expect the sorting function to be expecting a string., will fix

Reply:
all fixed now